docs/proposals/resourcebind-life-cycle/resourcebinding-behavior.md

```diff
@@ -0,0 +1,128 @@
+---
+title: Behavior Of ResourceBind When ProPagation Policy ' s ResourceSelector Change
+
+authors:
+- "@olderTaoist"
+
+reviewers:
+- "@XiShanYongYe - Chang"
+- "@weilaaa"
+
+approvers:
+
+creation-date: 2023-06-27
+
+---
+
+# Behavior Of ResourceBind When ProPagation Policy ' s ResourceSelector Change
+
+## Summary
+
+Current there are two situations that `ResourceBinding` will be deleted :
+- the releated resource template be deleted
+- the releated propagation policy be deleted
+  
+When change ResourceSelector of `ProPagationPolicy` or `ClusterProPagationPolicy`, The `ResourceBinding` of older 
+matched resource template is still exist , But users want `ResourceBinding` to be deleted in some scenario.
+  
+This proposal aims to provide a solution for `ResourceBinding` whether to be deleted depend on users.
+
+
+## Motivation
+
+Users want controller the behavior(delete or reserve) of ResourceBinding when change ResourceSelector 
+of `ProPagationPolicy` or `ClusterProPagationPolicy`
+
+### Goals
+
+- Provide a solution to custom behavior(delete or reserve) of `ResourceBinding` depend on users.
+
+
+### Non-Goals
+
+
+## Proposal
+
+
+### User Stories (Optional)
+
+
+#### As a user , I want resourcebinding to be deleted .
+
+There some resource in member clusters controller by karmada with a propagation policy. When i change the ResourceSelector
+of `ProPagationPolicy`, I want old resources(not match new ResourceSelector) in member clusters to be deleted,
+The current situation is the old resource still exist in member cluster.
+
+
+
+### Notes/Constraints/Caveats (Optional)
+
+### Risks and Mitigations
+
+## Design Details
+
+add `NotAllowOrphanRB` filed in `ResourceSelector` struct. User can use `NotAllowOrphanRB` field determinate whether to
+delete `ResourceBinding` when change `ResourceSelector` 
+
+### Older ResourceSelector Struct
+
+```go
+// ResourceSelector the resources will be selected.
+type ResourceSelector struct {
+	APIVersion    string `json:"apiVersion"`
+	Kind          string `json:"kind"`
+	Namespace     string `json:"namespace,omitempty"`
+	Name          string `json:"name,omitempty"`
+	LabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty"`
+}
+```
+
+### New ResourceSelector Struct
+
+```go
+// ResourceSelector the resources will be selected.
+type ResourceSelector struct {
+    APIVersion    string `json:"apiVersion"`
+    Kind          string `json:"kind"`
+    Namespace     string `json:"namespace,omitempty"`
+    Name          string `json:"name,omitempty"`
+    LabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty"`
+
+    // delete or reserve resourcebinding when change resource selector .
+    // Default is false , which means reserve resourcebinding .
+    // +optional 
+    NotAllowOrphanRB bool `json:"notAllowOrphanRB,omitempty"`
```

Comment:
As we talked at the [2023-08-15 Community meeting](https://docs.google.com/document/d/1y6YLVC-v7cmVAdbjedoyR5WL0-q45DBRXTvz5_I7bkA/edit?pli=1#heading=h.ltrvkc5ut73y), we might need a field in `.spec.` to declare the deletion behavior. 

I'm still thinking about how to define the behavior, but we should declare it from the user perspective.